website/project/mailing_list.py

```diff
@@ -0,0 +1,161 @@
+import re
+
+from flask import request
+from modularodm import Q
+
+from framework.auth.core import get_user
+from framework.auth.signals import user_confirmed
+
+from website import mails
+from website import settings
+from website.models import Node, NotificationSubscription
+from website.notifications.utils import to_subscription_key
+
+from website.project.model import MailingListEventLog
+from website.project.signals import contributor_added
+from website.util.sanitize import unescape_entities
+
+###############################################################################
+# Signalled Functions
+###############################################################################
+
+@contributor_added.connect
+def subscribe_contributor_to_mailing_list(node, contributor, auth=None):
+    if node.mailing_enabled and contributor.is_active:
+        subscription = node.get_or_create_mailing_list_subscription()
+        subscription.add_user_to_subscription(contributor, 'email_transactional', save=True)
+        subscription.save()
+
+@user_confirmed.connect
+def resubscribe_on_confirm(user):
+    for node in user.contributed:
+        subscribe_contributor_to_mailing_list(node, user)
+
+###############################################################################
+# Mailing List Functions
+###############################################################################
+
+def address(node_id):
+    return node_id + '@' + settings.SHORT_DOMAIN
+
+def find_email(long_email):
```

Comment:
There is similar code for this in the conferences module: https://github.com/CenterForOpenScience/osf.io/blob/92798bba8f8019ae8f53c3b6a06dc01ab5e3d7f7/website/conferences/message.py#L111-L115 ; can you DRY this up?